classes/class-wc-connect-shipping-label.php

```diff
@@ -312,12 +312,110 @@ protected function get_form_data( WC_Order $order ) {
 			return $form_data;
 		}
 
-		private function is_supported_country( $country_code ) {
-			return in_array( $country_code, $this->supported_countries );
+		/**
+		 * Check whether the given order is eligible for shipping label creation - the order has at least one product that is:
+		 * - Shippable.
+		 * - Non-refunded.
+		 * - Not already included in an existing non-refunded shipping label.
+		 *
+		 * @param WC_Order $order The order to check for shipping label creation eligibility.
+		 * @return bool Whether the given order is eligible for shipping label creation.
+		 */
+		public function is_order_eligible_for_shipping_label_creation( WC_Order $order ): bool {
+			// Set up a dictionary from product ID to quantity in the order, which will be updated by refunds and existing labels later.
+			$quantities_by_product_id = array();
+			foreach ( $order->get_items() as $item ) {
+				$product = WC_Connect_Compatibility::instance()->get_item_product( $order, $item );
+				if ( $product && $product->needs_shipping() ) {
+					$product_id                              = WC_Connect_Compatibility::instance()->get_product_id( $product );
+					$current_quantity                        = array_key_exists( $product_id, $quantities_by_product_id ) ? $quantities_by_product_id[ $product_id ] : 0;
+					$quantities_by_product_id[ $product_id ] = $current_quantity + $item->get_quantity();
+				}
+			}
+
+			// A shipping label cannot be created without a shippable product.
+			if ( empty( $quantities_by_product_id ) ) {
+				return false;
+			}
+
+			// Update the quantity for each refunded product ID in the order.
+			foreach ( $order->get_refunds() as $refund ) {
+				foreach ( $refund->get_items() as $refunded_item ) {
+					$product    = WC_Connect_Compatibility::instance()->get_item_product( $order, $refunded_item );
+					$product_id = WC_Connect_Compatibility::instance()->get_product_id( $product );
+					if ( array_key_exists( $product_id, $quantities_by_product_id ) ) {
+						$current_count                           = $quantities_by_product_id[ $product_id ];
+						$quantities_by_product_id[ $product_id ] = $current_count - abs( $refunded_item->get_quantity() );
+					}
+				}
+			}
+
+			// Update the quantity for each product ID that is in a non-refunded shipping label package.
```

Comment:
An order can have a label generated. If the label is unused, the user can refund it within 30 days. In WCS UI, it is possible that the user print multiple labels on the same order without refunding the previous ones first.

In the code below, as soon as a label is generated (even if it's not used), the `creation_eligibility` end point will return `"is_eligible": false,`. Just want to check with you if this is the expected behaviour.

Reply:
Good question! The current design in mobile is to exclude products that have been in a non-refunded shipping label, but it might be better to match the behavior with the web client. I just asked about this in CGPNUU63E/p1615285371106700-slack - will confirm!

Reply:
I dug into the log a bit. I think the button used to be called "Redo shipping" after a label is printed. https://github.com/Automattic/woocommerce-services/commit/81edffe4ee44d0f4fdcb80648692e59d518174ae

The idea of allowing multiple labels printed on the same order might have something to do with redoing fulfillment. I can see this happening when a shipment is missed and they have to reprint a new label.